onnxruntime/core/framework/tensorprotoutils.cc

```diff
@@ -541,7 +515,7 @@ TensorProto::DataType GetTensorProtoType(const Tensor& tensor) {
 ONNX_NAMESPACE::TensorProto TensorToTensorProto(const Tensor& tensor, const std::string& tensor_proto_name,
                                                 const ONNX_NAMESPACE::TypeProto& tensor_proto_type) {
   // Given we are using the raw_data field in the protobuf, this will work only for little-endian format.
-  ORT_ENFORCE(IsLittleEndianOrder());
+  ORT_ENFORCE(endian::native == endian::little);
```

Comment:
>ORT_ENFORCE [](start = 2, length = 11)

static_assert?

Reply:
That appears to be the case, as in the original version.